napari/layers/labels/labels.py

```diff
@@ -528,6 +533,8 @@ def num_colors(self):
 
     @num_colors.setter
     def num_colors(self, num_colors):
+        if num_colors < 1 or num_colors >= 2**16:
+            raise ValueError("num_colors must be between 1 and 2**16 - 1")
```

Comment:
```suggestion
            raise ValueError("num_colors must be between 1 and 65535")
```

Reply:
Doesn't have to be addressed in this PR but what did you think of my comment to deprecate num_colors? https://github.com/napari/napari/issues/6448#issuecomment-1812414837 I think it is redundant information...

Reply:
num_colors is only stable API for chnge colors between difference computers (as seed may change behaviour along time)

Reply:
But that can change if we use `colormap` as the API and provide specific labels colormaps — including named colormaps. With the new, simpler mapping that you are doing here (basically modulo), colormaps can behave much more sanely (this has been a long-standing request), essentially like color cycles. So we can have "lowdisc" (or something suitably catchy 😂) for the default napari colormap, "glasbey50" "glasbey255", etc. So a user can always do `labels_layer.colormap = 'lowdisc'` to get back to the original colormap.

It will also be really nice for things like reproducing colormaps in matplotlib etc.